packages/eslint-plugin-query/src/rules/exhaustive-deps.test.ts

```diff
@@ -0,0 +1,83 @@
+import { ESLintUtils } from '@typescript-eslint/utils'
+import { exhaustiveDepsRule } from './exhaustive-deps.rule'
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+  settings: {},
+})
+
+ruleTester.run('exhaustive-deps', exhaustiveDepsRule, {
+  valid: [
+    {
+      name: 'should pass when deps are passed in array',
+      code: 'useQuery({ queryKey: ["entity", id], queryFn: () => api.getEntity(id) });',
+    },
+    {
+      name: 'should pass when deps are passed in template literal',
+      // eslint-disable-next-line no-template-curly-in-string
+      code: 'useQuery({ queryKey: [`entity/${id}`], queryFn: () => api.getEntity(id) });',
+    },
+    {
+      name: 'should not pass fetch',
+      code: 'useQuery({ queryKey: ["entity", id], queryFn: () => fetch("...") });',
+    },
+    {
+      name: 'should not pass axios',
+      code: 'useQuery({ queryKey: ["entity", id], queryFn: () => axios.get("...") });',
+    },
+    {
+      name: 'should not pass api',
+      code: 'useQuery({ queryKey: ["entity", id], queryFn: () => api.get("...") });',
+    },
+  ],
+  invalid: [
+    {
+      name: 'should fail when no deps are passed',
+      code: `
+          const id = 1;
+          useQuery({ queryKey: ["entity"], queryFn: () => api.getEntity(id) });
+        `,
+      output: `
+          const id = 1;
+          useQuery({ queryKey: ["entity", id], queryFn: () => api.getEntity(id) });
+        `,
+      errors: [{ messageId: 'missingDeps', data: { deps: 'id' } }],
+    },
+    {
+      name: 'should fail when deps are passed incorrectly',
+      code: `
+            const id = 1;
+            useQuery({ queryKey: ["entity/\${id}"], queryFn: () => api.getEntity(id) });
+          `,
+      output: `
+            const id = 1;
+            useQuery({ queryKey: ["entity/\${id}", id], queryFn: () => api.getEntity(id) });
+          `,
+      errors: [{ messageId: 'missingDeps', data: { deps: 'id' } }],
+    },
+    {
+      name: 'should fail when dep exists inside setter and missing in queryKey',
+      code: `
+        const [id] = React.useState(1);
+        useQuery({
+            queryKey: ["entity"],
+            queryFn: () => {
+                const { data } = axios.get(\`.../\${id}\`);
+                return data;
+            }
+        });
+      `,
+      output: `
+        const [id] = React.useState(1);
+        useQuery({
```

Comment:
i can see that all tests use the object syntax for useQuery. is this a limitation that it will only work with that syntax, or does it also work for `useQuery(key, fn)` ?

(note that we want to move towards `useQuery(options)` in v5 anyways, so it wouldn't be a big deal)

Reply:
I haven't implemented the "traditional" syntax, but I guess I could write an edge case for this. But what if I could write another lint rule that will auto-fix "traditional" syntax into "object" syntax?

Reply:
This would be useful for transition from `v4` to `v5`, as instead of creating codemod, we could say `please install eslint plugin` for auto-fix.
This could also increase adoption rate.

Reply:
yeah this is actually a rule that @KubaJastrz has in his eslint-plugin:

- [prefer-object-syntax](https://github.com/KubaJastrz/eslint-plugin-react-query/blob/master/docs/rules/prefer-query-object-syntax.md)

and it's also fixable ❤️ 

since we got his permission to use it, maybe we should just copy it over and then it's totally fine that the exhaustive-deps rule only works on the object syntax. It will give people another incentive to switch towards that syntax, and it will also make the migration path to v5 better.